edgelet/edgelet-kube/src/module/create.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use futures::future::Either;
+use futures::prelude::*;
+use futures::{future, Future, Stream};
+use hyper::service::Service;
+use hyper::Body;
+
+use edgelet_docker::DockerConfig;
+use kube_client::{Error as KubeClientError, TokenSource};
+
+use crate::constants::EDGE_EDGE_AGENT_NAME;
+use crate::convert::{spec_to_deployment, spec_to_role_binding, spec_to_service_account};
+use crate::error::Error;
+use crate::runtime::KubeRuntimeData;
+use crate::KubeModuleRuntime;
+use edgelet_core::ModuleSpec;
+
+pub fn create_module<T, S>(
+    runtime: &KubeModuleRuntime<T, S>,
+    module: &ModuleSpec<DockerConfig>,
+) -> impl Future<Item = (), Error = Error>
+where
+    T: TokenSource + Send + 'static,
+    S: Service + Send + 'static,
+    S::ReqBody: From<Vec<u8>>,
+    S::ResBody: Stream,
+    Body: From<S::ResBody>,
+    S::Error: Into<KubeClientError>,
+    S::Future: Send,
+{
+    let runtime_for_sa = runtime.clone();
+    let module_for_sa = module.clone();
+
+    let runtime_for_deployment = runtime.clone();
+    let module_for_deployment = module.clone();
+
+    let inner = create_or_update_service_account(&runtime, &module)
+        .and_then(move |_| create_or_update_role_binding(&runtime_for_sa, &module_for_sa))
+        .and_then(move |_| {
+            create_or_update_deployment(&runtime_for_deployment, &module_for_deployment)
+        });
+
+    Box::new(inner)
```

Comment:
Is there a way to avoid the Box?

Reply:
I don't think so. Returning type is expected to be `Future<(),Error>` but here I return `Then<Then<...>>`. Any ideas?

Reply:
:(

Reply:
Will `flatten()` work here?

Reply:
I don't think you need any boxes in this file. AndThen implements IntoFuture.